### PR TITLE
Migrate IIS package

### PR DESCRIPTION
**Pull request by narph**
## What does this PR do?

- adds pipeline tests
- adds missing field definitions
- update ecs fields
- test metrics and logs (below examples)
- test dashboards and clean up text/labels

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md#tips-for-building-integrations) and this pull request is aligned with them.
- [x] I have verified that all datasets collect metrics or logs.

## Related issues

- Closes https://github.com/elastic/integrations/issues/518

## Screenshots

```
 {
        "_index" : ".ds-logs-iis.access-default-2021.01.19-000001",
        "_type" : "_doc",
        "_id" : "Ioy7GncBpTwhfSbiNm-w",
        "_score" : 0.0,
        "_source" : {
          "agent" : {
            "hostname" : "DESKTOP-K76UDQL",
            "name" : "DESKTOP-K76UDQL",
            "id" : "7b60956b-400e-44fa-8d0f-c8eac8eec6c6",
            "type" : "filebeat",
            "ephemeral_id" : "c8ad9ba9-6493-4b13-ab2e-92f97496f5bf",
            "version" : "7.11.0"
          },
          "temp" : { },
          "log" : {
            "file" : {
              "path" : """C:\inetpub\logs\LogFiles\W3SVC1\u_ex201207_x.log"""
            },
            "offset" : 261
          },
          "elastic_agent" : {
            "id" : "4061ca50-5a56-11eb-958d-53f77244394d",
            "version" : "7.11.0",
            "snapshot" : false
          },
          "destination" : {
            "address" : "::1",
            "port" : 80,
            "ip" : "::1"
          },
          "source" : {
            "address" : "::1",
            "ip" : "::1"
          },
          "url" : {
            "path" : "/"
          },
          "input" : {
            "type" : "log"
          },
          "iis" : {
            "access" : {
              "sub_status" : 0,
              "win32_status" : 0
            }
          },
          "@timestamp" : "2020-12-07T12:46:54.000Z",
          "ecs" : {
            "version" : "1.5.0"
          },
          "related" : {
            "ip" : [
              "::1",
              "::1"
            ]
          },
          "data_stream" : {
            "namespace" : "default",
            "type" : "logs",
            "dataset" : "iis.access"
          },
          "host" : {
            "hostname" : "DESKTOP-K76UDQL",
            "os" : {
              "build" : "19041.746",
              "kernel" : "10.0.19041.746 (WinBuild.160101.0800)",
              "name" : "Windows 10 Pro",
              "family" : "windows",
              "version" : "10.0",
              "platform" : "windows"
            },
            "ip" : [
              "2a02:a446:f25b:1:558d:72b2:74b6:7c67",
              "2a02:a446:f25b:1:81e9:c682:4947:746d",
              "fe80::558d:72b2:74b6:7c67",
              "192.168.2.10",
              "fe80::f09a:4481:bc48:e3e3",
              "169.254.227.227",
              "fe80::d135:57b1:9b7f:e10a",
              "169.254.225.10",
              "fe80::6901:1ee0:60dd:c986",
              "169.254.201.134",
              "fe80::fc98:6ed8:6d25:3d98",
              "169.254.61.152",
              "fe80::e821:c725:2d77:e33f",
              "172.31.128.1",
              "fe80::4186:d0e9:2883:74c4",
              "172.17.192.1",
              "fe80::dcd3:95bf:eb18:33b",
              "172.20.224.1"
            ],
            "name" : "DESKTOP-K76UDQL",
            "id" : "bc395ab9-1a1a-4312-ad0b-a05cceaf78f1",
            "mac" : [
              "cc:48:3a:b9:87:c0",
              "9c:b6:d0:3e:a8:c4",
              "9e:b6:d0:3e:a8:c4",
              "ae:b6:d0:3e:a8:c4",
              "28:c2:1f:52:ff:bf",
              "00:15:5d:68:b8:53",
              "00:15:5d:73:e7:06",
              "00:15:5d:fe:ee:90"
            ],
            "architecture" : "x86_64"
          },
          "http" : {
            "request" : {
              "method" : "GET"
            },
            "response" : {
              "status_code" : 200
            }
          },
          "event" : {
            "duration" : 169000000,
            "ingested" : "2021-01-19T13:00:12.479633200Z",
            "created" : "2021-01-19T13:00:12.001Z",
            "kind" : "event",
            "category" : [
              "web",
              "network"
            ],
            "type" : [
              "connection"
            ],
            "dataset" : "iis.access",
            "outcome" : "success"
          },
          "user_agent" : {
            "original" : "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.67 Safari/537.36 Edg/87.0.664.47",
            "os" : {
              "name" : "Windows",
              "version" : "10",
              "full" : "Windows 10"
            },
            "name" : "Edge",
            "device" : {
              "name" : "Other"
            },
            "version" : "87.0.664.47"
          }
        }
      }
```

Error logs:

```
    "hits" : [
      {
        "_index" : ".ds-logs-iis.error-default-2021.01.19-000001",
        "_type" : "_doc",
        "_id" : "94y7GncBpTwhfSbiNm6w",
        "_score" : 0.0,
        "_source" : {
          "agent" : {
            "hostname" : "DESKTOP-K76UDQL",
            "name" : "DESKTOP-K76UDQL",
            "id" : "7b60956b-400e-44fa-8d0f-c8eac8eec6c6",
            "ephemeral_id" : "c8ad9ba9-6493-4b13-ab2e-92f97496f5bf",
            "type" : "filebeat",
            "version" : "7.11.0"
          },
          "log" : {
            "file" : {
              "path" : """c:\Windows\System32\LogFiles\HTTPERR\httperr1.log"""
            },
            "offset" : 221
          },
          "elastic_agent" : {
            "id" : "4061ca50-5a56-11eb-958d-53f77244394d",
            "version" : "7.11.0",
            "snapshot" : false
          },
          "destination" : {
            "address" : "127.0.0.1",
            "port" : 80,
            "ip" : "127.0.0.1"
          },
          "source" : {
            "address" : "127.0.0.1",
            "port" : 61251,
            "ip" : "127.0.0.1"
          },
          "input" : {
            "type" : "log"
          },
          "iis" : {
            "error" : {
              "queue_name" : "Timer_ConnectionIdle"
            }
          },
          "@timestamp" : "2021-01-13T16:00:39.000Z",
          "ecs" : {
            "version" : "1.5.0"
          },
          "related" : {
            "ip" : [
              "127.0.0.1",
              "127.0.0.1"
            ]
          },
          "data_stream" : {
            "namespace" : "default",
            "type" : "logs",
            "dataset" : "iis.error"
          },
          "host" : {
            "hostname" : "DESKTOP-K76UDQL",
            "os" : {
              "build" : "19041.746",
              "kernel" : "10.0.19041.746 (WinBuild.160101.0800)",
              "name" : "Windows 10 Pro",
              "family" : "windows",
              "version" : "10.0",
              "platform" : "windows"
            },
            "ip" : [
              "2a02:a446:f25b:1:558d:72b2:74b6:7c67",
              "2a02:a446:f25b:1:81e9:c682:4947:746d",
              "fe80::558d:72b2:74b6:7c67",
              "192.168.2.10",
              "fe80::f09a:4481:bc48:e3e3",
              "169.254.227.227",
              "fe80::d135:57b1:9b7f:e10a",
              "169.254.225.10",
              "fe80::6901:1ee0:60dd:c986",
              "169.254.201.134",
              "fe80::fc98:6ed8:6d25:3d98",
              "169.254.61.152",
              "fe80::e821:c725:2d77:e33f",
              "172.31.128.1",
              "fe80::4186:d0e9:2883:74c4",
              "172.17.192.1",
              "fe80::dcd3:95bf:eb18:33b",
              "172.20.224.1"
            ],
            "name" : "DESKTOP-K76UDQL",
            "id" : "bc395ab9-1a1a-4312-ad0b-a05cceaf78f1",
            "mac" : [
              "cc:48:3a:b9:87:c0",
              "9c:b6:d0:3e:a8:c4",
              "9e:b6:d0:3e:a8:c4",
              "ae:b6:d0:3e:a8:c4",
              "28:c2:1f:52:ff:bf",
              "00:15:5d:68:b8:53",
              "00:15:5d:73:e7:06",
              "00:15:5d:fe:ee:90"
            ],
            "architecture" : "x86_64"
          },
          "event" : {
            "created" : "2021-01-19T13:00:12.000Z",
            "kind" : "event",
            "category" : [
              "web",
              "network"
            ],
            "type" : [
              "connection"
            ],
            "dataset" : "iis.error"
          }
        }
      }
```

Metrics:

application pool:

```
      {
        "_index" : ".ds-metrics-iis.application_pool-default-2021.01.19-000001",
        "_type" : "_doc",
        "_id" : "Poy8GncBpTwhfSbicehJ",
        "_score" : 1.0,
        "_source" : {
          "@timestamp" : "2021-01-19T13:01:32.677Z",
          "process" : {
            "pid" : 24688
          },
          "elastic_agent" : {
            "version" : "7.11.0",
            "id" : "4061ca50-5a56-11eb-958d-53f77244394d",
            "snapshot" : false
          },
          "ecs" : {
            "version" : "1.7.0"
          },
          "event" : {
            "module" : "iis",
            "duration" : 197520100,
            "dataset" : "iis.application_pool"
          },
          "data_stream" : {
            "type" : "metrics",
            "dataset" : "iis.application_pool",
            "namespace" : "default"
          },
          "host" : {
            "name" : "DESKTOP-K76UDQL",
            "hostname" : "DESKTOP-K76UDQL",
            "architecture" : "x86_64",
            "os" : {
              "family" : "windows",
              "name" : "Windows 10 Pro",
              "kernel" : "10.0.19041.746 (WinBuild.160101.0800)",
              "build" : "19041.746",
              "platform" : "windows",
              "version" : "10.0"
            },
            "id" : "bc395ab9-1a1a-4312-ad0b-a05cceaf78f1",
            "ip" : [
              "2a02:a446:f25b:1:558d:72b2:74b6:7c67",
              "2a02:a446:f25b:1:81e9:c682:4947:746d",
              "fe80::558d:72b2:74b6:7c67",
              "192.168.2.10",
              "fe80::f09a:4481:bc48:e3e3",
              "169.254.227.227",
              "fe80::d135:57b1:9b7f:e10a",
              "169.254.225.10",
              "fe80::6901:1ee0:60dd:c986",
              "169.254.201.134",
              "fe80::fc98:6ed8:6d25:3d98",
              "169.254.61.152",
              "fe80::e821:c725:2d77:e33f",
              "172.31.128.1",
              "fe80::4186:d0e9:2883:74c4",
              "172.17.192.1",
              "fe80::dcd3:95bf:eb18:33b",
              "172.20.224.1"
            ],
            "mac" : [
              "cc:48:3a:b9:87:c0",
              "9c:b6:d0:3e:a8:c4",
              "9e:b6:d0:3e:a8:c4",
              "ae:b6:d0:3e:a8:c4",
              "28:c2:1f:52:ff:bf",
              "00:15:5d:68:b8:53",
              "00:15:5d:73:e7:06",
              "00:15:5d:fe:ee:90"
            ]
          },
          "agent" : {
            "id" : "0ea27417-2d87-4abe-9409-58817ef3d564",
            "name" : "DESKTOP-K76UDQL",
            "type" : "metricbeat",
            "version" : "7.11.0",
            "hostname" : "DESKTOP-K76UDQL",
            "ephemeral_id" : "5477ee39-f339-465d-a804-89156e9890e4"
          },
          "metricset" : {
            "name" : "application_pool",
            "period" : 10000
          },
          "service" : {
            "type" : "iis"
          },
          "iis" : {
            "application_pool" : {
              "process" : {
                "private_bytes" : 1.0756096E8,
                "virtual_bytes" : 2.22527889408E12,
                "thread_count" : 62,
                "io_write_operations_per_sec" : 2.6598547065203006,
                "working_set" : 4.583424E7,
                "io_read_operations_per_sec" : 2.6598547065203006,
                "page_faults_per_sec" : 0.7881050982282373,
                "cpu_usage_perc" : 0,
                "handle_count" : 1831
              },
              "name" : "DefaultAppPool",
              "net_clr" : {
                "total_exceptions_thrown" : 0,
                "exceptions_thrown_per_sec" : 0,
                "finallys_per_sec" : 0,
                "filters_per_sec" : 0,
                "throw_to_catch_depth_per_sec" : 0
              }
            }
          }
        }
      },
```

webserver

```
{
        "_index" : ".ds-metrics-iis.webserver-default-2021.01.19-000001",
        "_type" : "_doc",
        "_id" : "NYy7GncBpTwhfSbihufc",
        "_score" : 1.0,
        "_source" : {
          "@timestamp" : "2021-01-19T13:00:32.698Z",
          "agent" : {
            "id" : "0ea27417-2d87-4abe-9409-58817ef3d564",
            "name" : "DESKTOP-K76UDQL",
            "type" : "metricbeat",
            "version" : "7.11.0",
            "hostname" : "DESKTOP-K76UDQL",
            "ephemeral_id" : "5477ee39-f339-465d-a804-89156e9890e4"
          },
          "event" : {
            "dataset" : "iis.webserver",
            "module" : "iis",
            "duration" : 10000200
          },
          "metricset" : {
            "name" : "webserver",
            "period" : 10000
          },
          "service" : {
            "type" : "iis"
          },
          "data_stream" : {
            "dataset" : "iis.webserver",
            "namespace" : "default",
            "type" : "metrics"
          },
          "ecs" : {
            "version" : "1.7.0"
          },
          "host" : {
            "id" : "bc395ab9-1a1a-4312-ad0b-a05cceaf78f1",
            "name" : "DESKTOP-K76UDQL",
            "ip" : [
              "2a02:a446:f25b:1:558d:72b2:74b6:7c67",
              "2a02:a446:f25b:1:81e9:c682:4947:746d",
              "fe80::558d:72b2:74b6:7c67",
              "192.168.2.10",
              "fe80::f09a:4481:bc48:e3e3",
              "169.254.227.227",
              "fe80::d135:57b1:9b7f:e10a",
              "169.254.225.10",
              "fe80::6901:1ee0:60dd:c986",
              "169.254.201.134",
              "fe80::fc98:6ed8:6d25:3d98",
              "169.254.61.152",
              "fe80::e821:c725:2d77:e33f",
              "172.31.128.1",
              "fe80::4186:d0e9:2883:74c4",
              "172.17.192.1",
              "fe80::dcd3:95bf:eb18:33b",
              "172.20.224.1"
            ],
            "mac" : [
              "cc:48:3a:b9:87:c0",
              "9c:b6:d0:3e:a8:c4",
              "9e:b6:d0:3e:a8:c4",
              "ae:b6:d0:3e:a8:c4",
              "28:c2:1f:52:ff:bf",
              "00:15:5d:68:b8:53",
              "00:15:5d:73:e7:06",
              "00:15:5d:fe:ee:90"
            ],
            "hostname" : "DESKTOP-K76UDQL",
            "architecture" : "x86_64",
            "os" : {
              "build" : "19041.746",
              "platform" : "windows",
              "version" : "10.0",
              "family" : "windows",
              "name" : "Windows 10 Pro",
              "kernel" : "10.0.19041.746 (WinBuild.160101.0800)"
            }
          },
          "iis" : {
            "webserver" : {
              "network" : {
                "post_requests_per_sec" : 0,
                "service_uptime" : 561984,
                "get_requests_per_sec" : 0,
                "bytes_sent_per_sec" : 0,
                "anonymous_users_per_sec" : 0,
                "bytes_received_per_sec" : 0,
                "maximum_connections" : 11,
                "total_bytes_sent" : 5407598.0,
                "current_anonymous_users" : 0,
                "current_connections" : 0,
                "total_get_requests" : 130,
                "total_delete_requests" : 0,
                "total_bytes_received" : 63084,
                "total_non_anonymous_users" : 0,
                "current_non_anonymous_users" : 0,
                "total_connection_attempts" : 43,
                "total_post_requests" : 0,
                "total_anonymous_users" : 130,
                "delete_requests_per_sec" : 0
              },
              "cache" : {
                "file_cache_hits" : 7,
                "maximum_file_cache_memory_usage" : 99453,
                "total_files_cached" : 5,
                "output_cache_total_hits" : 0,
                "output_cache_current_memory_usage" : 0,
                "output_cache_total_misses" : 138,
                "current_files_cached" : 3,
                "total_uris_cached" : 43,
                "file_cache_misses" : 28,
                "current_uris_cached" : 0,
                "output_cache_current_items" : 0,
                "uri_cache_hits" : 41,
                "current_file_cache_memory_usage" : 99453,
                "uri_cache_misses" : 97
              },
              "process" : {
                "private_bytes" : 2.4713216E8,
                "worker_process_count" : 3,
                "working_set" : 1.70098688E8,
                "virtual_bytes" : 4.45123209216E12,
                "cpu_usage_perc" : 0,
                "handle_count" : 3292,
                "io_write_operations_per_sec" : 2.7281720319807237,
                "io_read_operations_per_sec" : 2.728172031980724,
                "thread_count" : 162,
                "page_faults_per_sec" : 0.8083472687350293
              },
              "asp_net" : {
                "application_restarts" : 0,
                "request_wait_time" : 0
              },
              "asp_net_application" : {
                "errors_total_per_sec" : 0,
                "requests_per_sec" : 0,
                "requests_executing" : 0,
                "pipeline_instance_count" : 1,
                "requests_in_application_queue" : 0
              }
            }
          },
          "elastic_agent" : {
            "version" : "7.11.0",
            "id" : "4061ca50-5a56-11eb-958d-53f77244394d",
            "snapshot" : false
          }
        }
      },
```

website
```
 {
        "_index" : ".ds-metrics-iis.website-default-2021.01.19-000001",
        "_type" : "_doc",
        "_id" : "4Iy8GncBpTwhfSbiI-c4",
        "_score" : 1.0,
        "_source" : {
          "@timestamp" : "2021-01-19T13:01:12.568Z",
          "service" : {
            "type" : "iis"
          },
          "iis" : {
            "website" : {
              "name" : "atest.local",
              "network" : {
                "service_uptime" : 562024,
                "bytes_sent_per_sec" : 0,
                "current_connections" : 0,
                "total_get_requests" : 67,
                "total_connection_attempts" : 20,
                "maximum_connections" : 5,
                "total_post_requests" : 0,
                "total_put_requests" : 0,
                "total_bytes_sent" : 2907829.0,
                "total_bytes_received" : 29006,
                "get_requests_per_sec" : 0,
                "delete_requests_per_sec" : 0,
                "total_delete_requests" : 0,
                "put_requests_per_sec" : 0,
                "post_requests_per_sec" : 0,
                "bytes_received_per_sec" : 0
              }
            }
          },
          "data_stream" : {
            "dataset" : "iis.website",
            "namespace" : "default",
            "type" : "metrics"
          },
          "host" : {
            "os" : {
              "kernel" : "10.0.19041.746 (WinBuild.160101.0800)",
              "build" : "19041.746",
              "platform" : "windows",
              "version" : "10.0",
              "family" : "windows",
              "name" : "Windows 10 Pro"
            },
            "id" : "bc395ab9-1a1a-4312-ad0b-a05cceaf78f1",
            "ip" : [
              "2a02:a446:f25b:1:558d:72b2:74b6:7c67",
              "2a02:a446:f25b:1:81e9:c682:4947:746d",
              "fe80::558d:72b2:74b6:7c67",
              "192.168.2.10",
              "fe80::f09a:4481:bc48:e3e3",
              "169.254.227.227",
              "fe80::d135:57b1:9b7f:e10a",
              "169.254.225.10",
              "fe80::6901:1ee0:60dd:c986",
              "169.254.201.134",
              "fe80::fc98:6ed8:6d25:3d98",
              "169.254.61.152",
              "fe80::e821:c725:2d77:e33f",
              "172.31.128.1",
              "fe80::4186:d0e9:2883:74c4",
              "172.17.192.1",
              "fe80::dcd3:95bf:eb18:33b",
              "172.20.224.1"
            ],
            "mac" : [
              "cc:48:3a:b9:87:c0",
              "9c:b6:d0:3e:a8:c4",
              "9e:b6:d0:3e:a8:c4",
              "ae:b6:d0:3e:a8:c4",
              "28:c2:1f:52:ff:bf",
              "00:15:5d:68:b8:53",
              "00:15:5d:73:e7:06",
              "00:15:5d:fe:ee:90"
            ],
            "hostname" : "DESKTOP-K76UDQL",
            "name" : "DESKTOP-K76UDQL",
            "architecture" : "x86_64"
          },
          "agent" : {
            "type" : "metricbeat",
            "version" : "7.11.0",
            "hostname" : "DESKTOP-K76UDQL",
            "ephemeral_id" : "5477ee39-f339-465d-a804-89156e9890e4",
            "id" : "0ea27417-2d87-4abe-9409-58817ef3d564",
            "name" : "DESKTOP-K76UDQL"
          },
          "event" : {
            "dataset" : "iis.website",
            "module" : "iis",
            "duration" : 987800
          },
          "metricset" : {
            "name" : "website",
            "period" : 10000
          },
          "elastic_agent" : {
            "version" : "7.11.0",
            "id" : "4061ca50-5a56-11eb-958d-53f77244394d",
            "snapshot" : false
          },
          "ecs" : {
            "version" : "1.7.0"
          }
        }
      },
```
image

image

image

image

### CODE REVIEW

File: packages/iis/data_stream/application_pool/manifest.yml [hunk 1]
```diff
@@ -5,3 +5,10 @@ streams:
   - input: iis/metrics
     title: IIS application_pool metrics
     description: Collect IIS application_pool metrics
```
Comment (mtojek) [inline packages/iis/data_stream/application_pool/manifest.yml:7]: I know it's not part of this PR, but I think you can remove these underscores.

File: packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+    "expected": [
+        {
+            "message": "#Software: Microsoft HTTP API 2.0",
+            "error": {
+                "message": "Provided Grok expressions do not match field value: [#Software: Microsoft HTTP API 2.0]"
```
Comment (mtojek) [inline packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json:6]: Hm.. something is wrong here. Did you expect these error messages here?

File: packages/iis/data_stream/application_pool/manifest.yml (hunk 1 above)
Comment (narph) [reply packages/iis/data_stream/application_pool/manifest.yml]: done

File: packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json (hunk 2 above)
Comment (narph) [reply packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json:6]: no, we filter those out but the configuration setting is found in `log.yml.hbs` file. Those settings do not apply in the pipeline tests.

Comment (mtojek) [reply packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json:6]: In this case I think it's worth stripping these entries from pipeline tests and leave only ones that can produce correct documents.

**Review by mtojek — COMMENTED** [review]
left comment

File: packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json (hunk 2 above)
Comment (narph) [reply packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json]: agree, it is for the best, will add in the future tests that will cover this scenario, I have updated the logs

**Review by mtojek — APPROVED** [review]
Ship it!

File: packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json (hunk 2 above)
Comment (jsoriano) [reply packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json]: Could you please open an issue about this? I think we should have tests covering processors in Filebeat working together with ingest pipelines.

Comment (mtojek) [reply packages/iis/data_stream/error/_dev/test/pipeline/test-ipv6-zone-id.log-expected.json]: They are covered with system tests, where you're spawning an instance of the Elastic Agent with filebeat/metricbeat and assigns a policy.

**Pull request merged**